### DISCUSSION TITLE
MQTT help for Jeedom Automation

### DISCUSSION THREAD

**Discussion by Vincent62100** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#548] [General]
Hi all, I'm french guy, so sorry in advance for my translation.
I need some help about MQTT issue with my french automation "Jeedom" (like home assistant).
If i try to add the "layzspa/command" with the value"4" , the pump does'nt start, MQTT works because I can receive some informations like:
 button NOBTN
command 4
cmd:4 4
message {"CONTENT":"STATES","TIME":1687164825,"LCK":1,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"CH1":32,"CH2":50,"CH3":57,"HJT":0,"BRT":7,"ERR":0,"GOD":0,"TGT":36,"TMP":29,"VTMC":29,"VTMF":84.19999695,"AMBC":20,"AMBF":68,"AMB":20,"VTM":29,"TGTC":36,"TMPC":29,"TGTF":97,"TMPF":84}
MQTT_Connect_Count 1
reboot_reason Power On
Status Alive 

Please see the informations receive by MQTT, and the last command added by me:
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/137434167/de13fc51-607e-42be-99ad-b0f8907287a5)
Thanks in advance for your help

**Comment by visualapproach** [comment]
Hi! I bet your English is way better than my French! Instead of just a '4' you need to send a json formatted string like (this is from my memory only and may contain errors)
{CMD:4, VALUE:1, XTIME: 0, INTERVAL: 0, TXT: "hello"}

If you look in ha.txt file in the code, you can see how it's formatted.

**Comment by Vincent62100** [comment]
Thanks, I tried with this string: {CMD:4,VALUE:1,XTIME:0,INTERVAL:0,TXT:"ON"} but nothing happens, I'm not at home and I asked to my wife to check, she's said that the pump does'nt start, I will continue trials when I will come back (next week), Now I will try to read documentation about JSON string.
So i will keep you updated later about my issue.
Thanks again for your support.
Best regards
Vincent

**Reply by visualapproach** [reply]
Ok I checked my HA and it should be citation marks around the keys:

{"CMD":4, ...

**Comment by domoChris** [comment]
Hello Vincent,

Here are some commands with jeedom and jMQTT

![spa_jeedom](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/42420158/d8deb38b-ad03-4171-b316-e1e84e547967)

Take care that my topic is not labeled as yours (spa vs layzspa)

**Reply by Vincent62100** [reply]
Hi Domo Chris, I don't understand, because i put the good commands but nothing happens by jMQTT,and if I try directly by wifi on 192.168.0.43 (Ip adress of my Lay-Z-Spa Module, buttons worrking well). 
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/137434167/bb5f1667-3900-4274-966f-d9e5682c0f70)
Thanks for your help

**Reply by Vincent62100** [reply]
@domoChris  what is the purpose of "set heure cible"  and "set temperature extérieure"? 
Used in script ? (scenario jeedom) ?
Do you have others command
Thanks for your reply
Vincent

**Comment by Vincent62100** [comment]
Hi Domo Chris and Visual Approch, the issue is solved, mistake from my side inside MQTT parameters, I put the wrong IP adress (IP layzspa) I replaced it by my automation ( IP Jeedom), now jMQTT request working well.
Thanks All for your support.
Have a good day
